server/s3_bucket_metadata_cache.cc

```diff
@@ -45,53 +45,61 @@ S3BucketMetadataCache::Item::Item(const S3BucketMetadata& src)
 }
 
 S3BucketMetadataCache::Item::~Item() {
-  assert(can_remove());
-  assert(fetch_waiters.empty());
-  assert(!on_changed);
+  if (!can_remove()) {
+    s3_log(S3_LOG_FATAL, nullptr, "Motr operation is in progress");
```

Comment:
which scenario this case can arise ?

Reply:
It's for detecting possible logic errors in future

Reply:
we need to give more info in the log message, to figure out what went wrong... 

_"Motr operation is in progress" is not enough._

Give info like _"during X, Y happened, which is not expected. Please check or fix Z."_

Reply:
More message can be given, which is good

better not to say to "fix Z" unless we know exactly how that can be fixed.